### ISSUE TITLE
depth>=3 doesn't work for self-related content

### ISSUE THREAD

**Issue by oarriet** [dotCMS/core#19796]
Reproducible in latest demo 21.01

## Steps to reproduce
1. Define content type "SelfJoin" which has a relationship pointed to "SelfJoin" itself.
![SelfJoin1](https://user-images.githubusercontent.com/944787/104363983-920f9d80-54db-11eb-8de2-ab9fd67740fa.png)
![SelfJoin2](https://user-images.githubusercontent.com/944787/104363984-9340ca80-54db-11eb-85c7-5b429918af98.png)
![SelfJoin3](https://user-images.githubusercontent.com/944787/104363987-93d96100-54db-11eb-8581-9f025f03dbba.png)

2. Create specific contents named "father","child" and "grandchild" of SelfJoin type. And father relates to a child, a child relates to grandchild.
![father](https://user-images.githubusercontent.com/944787/104363997-9936ab80-54db-11eb-9f3b-4acd6a1ec5cd.png)
![child](https://user-images.githubusercontent.com/944787/104364000-9b006f00-54db-11eb-9e5d-69b5bec2e474.png)

3. Press JSON under History of "father"

4.  Add /depth/3 at the tail of URL, for example: http://localhost:8080/api/content/id/b63bf70e-b6ae-42c2-9d3d-d024a12af53d/depth/3

5. "grandchild" should be retrieved out, but it doesn't.
```
{
  "contentlets": [
    {
      "publishDate": "2021-01-12 19:27:35.77",
      "inode": "273928e6-0f90-49bf-94c2-f8d163b7f490",
      "host": "48190c8c-42c4-46af-8d1a-0cd5db894797",
      "locked": false,
      "stInode": "6cebc911-3683-43da-bfd4-008eb0a95d61",
      "contentType": "SelfJoin",
      "identifier": "928f8806-51bd-456e-8ac6-4da594570922",
      "folder": "SYSTEM_FOLDER",
      "hasTitleImage": false,
      "sortOrder": 0,
      "hostName": "demo.dotcms.com",
      "modDate": "2021-01-12 19:27:35.77",
      "title": "father",
      "baseType": "CONTENT",
      "archived": false,
      "working": true,
      "live": true,
      "owner": "dotcms.org.1",
      "languageId": 1,
      "url": "/content.d2ffdafd-66f2-4988-b0fe-7a09c79e9a8d",
      "titleImage": "TITLE_IMAGE_NOT_FOUND",
      "modUserName": "Admin User",
      "hasLiveVersion": true,
      "modUser": "dotcms.org.1",
      "relateToSelf": [
        {
          "hostName": "demo.dotcms.com",
          "modDate": "2021-01-12 19:27:45.854",
          "publishDate": "2021-01-12 19:27:45.854",
          "title": "child",
          "baseType": "CONTENT",
          "inode": "caa60c9c-8361-42e0-a4e4-7e2a08a341c6",
          "archived": false,
          "host": "48190c8c-42c4-46af-8d1a-0cd5db894797",
          "working": true,
          "locked": false,
          "stInode": "6cebc911-3683-43da-bfd4-008eb0a95d61",
          "contentType": "SelfJoin",
          "live": true,
          "owner": "dotcms.org.1",
          "identifier": "b95a8aaa-f624-45d6-9142-46ff288a0d33",
          "languageId": 1,
          "url": "/content.0a1ed558-1d23-4689-84f6-e527ae93f85c",
          "titleImage": "TITLE_IMAGE_NOT_FOUND",
          "modUserName": "Admin User",
          "hasLiveVersion": true,
          "folder": "SYSTEM_FOLDER",
          "hasTitleImage": false,
          "sortOrder": 0,
          "modUser": "dotcms.org.1"
        }
      ]
    }
  ]
}
```

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/19882
FYI: max depth value is 3.

**Comment by bryanboza** [comment]
Fixed, tested on release-21.03 // Postgres // FF
